### PR TITLE
fix: Improve dialer pad and fix: app crash on number with char "+"

### PR DESCRIPTION
**Pull request by xdubx**
fix: https://github.com/Chooloo/koler/issues/441 fix out of bound error by updating the editText
fix: https://github.com/Chooloo/koler/issues/437 delete and typing in the editText now works with the cursor position 
fix:  https://github.com/Chooloo/koler/issues/486 replace the delete icon with the correct one, add missing description

### CODE REVIEW

File: chooloolib/src/main/java/com/chooloo/www/chooloolib/ui/dialpad/DialpadFragment.kt [hunk 1]
```diff
@@ -34,6 +34,9 @@ open class DialpadFragment @Inject constructor() : BaseFragment<DialpadViewState
                 addTextContextMenuItemListener(this@DialpadFragment)
             }
 
+            //promote edit to the viewstate
+            viewState.editTextRef.value = dialpadEditText
```
Comment (roeiedri) [inline chooloolib/src/main/java/com/chooloo/www/chooloolib/ui/dialpad/DialpadFragment.kt:38]: Please remove any view references from the viewstate
The view should listen for changes and act accordingly

**Review by roeiedri — CHANGES_REQUESTED** [review]
Please fix those
If it's an issue ill just merge and change it myself

**Comment by xdubx** [comment]
Yes I take a look an correct it.

**Comment by xdubx** [comment]
2/3 are now fixed? I close the pr

**Pull request closed without merging**